tox.ini

```diff
@@ -1,24 +1,31 @@
 [tox]
 envlist =
-    py3{11,10,9,8,7},pypy3{8,7}
+    py3{12,11,10,9,8,7}
+    pypy39
     style
     typing
     docs
 skip_missing_interpreters = true
 
 [testenv]
+package = wheel
+wheel_build_env = .pkg
```

Comment:
Are these related to the versions?

Reply:
This is some new tox 4 config so that a single wheel is built for all test envs, rather than building and installing the sdist each time.